pkg/datasources/account_roles.go

```diff
@@ -0,0 +1,94 @@
+package datasources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var accountRolesSchema = map[string]*schema.Schema{
+	"pattern": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Filters the command output by object name.",
+	},
```

Comment:
are we not doing in_class or starts_with?

Reply:
check my other comment about this change's purpose